### ISSUE TITLE
Missing Taxonomy

### ISSUE THREAD

**Issue by mycojon** [benjjneb/dada2#1822]
I have an issue that I'm assuming from dada2.

I have been running numerous 16s samples through a commercial lab.    They use the dada2 program for their bioinformatics.

Example.  Swab 1, run with V1V2, V3V4 (Illumina Miseq 2x300), and Pacbio V1-V9.   I had a specific expectation for Mycobacterium to be present in the samples, but none returned with the results.   I'm not a pro, but the read quality looks pretty good.

Ran the rawdata (various other samples) through Ezbiocloud (occasionally will identify Mycobacterium)
Ran through CZID.org, consistently finding Mycobacterium.  But it's generally only identifying Mycobacterium from the NCBI nr protein translated database.   Often reads are in the 10,000-40,000 range aligning to the the Mycobacterium species.

I have tried another approach to validate this issue.  Taking multiple samples from different location from the same patient at the same time concatenation of multiple samples so there are many more reads to work from.  Ezbiocloud easily identifies Mycobacterium at this point.

I'm trying to figure out what is causing the discrepancy.  
1.  Is it the analysis side in which the DNA are not clean enough ( have heard Mycobacterium have lysis issues)??  
2. Are there gaps in the reads (not uncommon for Mycobacterium I believe), so dada2 is rejecting the reads??
3. Is there a reason that Mycobacterium is primarily only being identified from the NCBI translated proteins database??
4.  There generally aren't any contigs identified shown on the CZID.org, does this mean that the paired end reads couldn't match up for some reason??

**Comment by benjjneb** [comment]
I can't tell you why this is happening. I can guess though, and my guess is that the reference database is 90% of the difference between different-but-reasonable taxonomic assignment methods.

**Comment by mycojon** [comment]
Do you have any suggestions???   Ideally I would like the analysis lab to be able to identify Mycobacterium in their results (if they are actually in the sample).   I'm not 100% sure if Mycobacterium is in the samples.
I'm running the samples through Zymo Research, which uses dada2.  They have run V1V2, V3V4, V1-V9.   Often results will have a significant amount of "other" bacteria not identified 8-12%, maybe the Mycobacterium is within that??

I'm not sure how the dada2 works.  Can it handle concatenating the V1-Forward + V3-Forward + V1-V9, essentially combining all of the best reads and skipping trying to merge any reverse reads.

Also, on how the CZID.org platform only identifies Mycobacterium on a small region of the protein database is this because the reads don't join??

Any suggestions??

Thanks

**Comment by benjjneb** [comment]
> I'm not sure how the dada2 works. Can it handle concatenating the V1-Forward + V3-Forward + V1-V9, essentially combining all of the best reads and skipping trying to merge any reverse reads.

I'm not even sure what that would mean, how would you concatenate reads from distinct sequencing datasets? Which reads e.g. V1 reads do you concatenate with which V3 reads?

> Also, on how the CZID.org platform only identifies Mycobacterium on a small region of the protein database is this because the reads don't join??

How are 16S reads being classified against a protein database? 16S doesn't encode a protein, its ribosomal RNA.

> Any suggestions??

Like I said above, my first guess would be differences in the reference databases. But to make progress on something like this you need to work up from simple test cases. Like a synthetic set of reads that are created from Mycobacteria, that are then classified through the different pipelines/reference-dbs you are trying to evaluate.

**Comment by mycojon** [comment]
Sorry, just learning stuff here.    I understand on the concatenating issue.

Maybe you have an idea on this issue.   On the CZID platform (uses BLASTn & BLASTx), say 16s V1V2 for example as well as V3V4.   It will match up the samples to the NCBI database.   It runs against the Nucleotide NT database, and also to the Translated Protein NR database.    My theory is that because Mycobacterium is a High GC bacteria analyzed on Illumina Miseq, is that reads are either trunicated or read quality falls off on those reads (this is a known issue for Illumina Reads with high GC content).   Because of this, it can’t join paired end reads.  Since the Nucleotide database wants perfect reads, it won’t align on there.  But when translated to the protein database BLASTX, the portion of the read that is good matches Mycobacterium Abscessus.   The results are about the same when either run as forward and reverse, or just the forward or reverse reads.

On my trial, I did a Primer Blast for Mycobacterium Abscessus.   Rather than use the primers, I took the region between the primers and trimmed my reads to that region.   Granted, it took the reads from all of the bacteria and did the same, but my focus was on the Mycobacterium Abscessus.   So running this again, BLASTX gave me a perfect match ID to the short length of the bacteria specific primer region.
On the same trial, concatenating multiple samples from the same person of just the V1V2 or V1V2 & V3V4 (not trimmed) all together.   Then I get the reads to align on both the NCBI NT & NR.

On dada2, I believe because of the high GC, trunicated reads, read quality dropping off, etc.,   it rejects all of those reads?  At least on the Miseq.

Does this make sense???
